idletiming_conn.go

```diff
@@ -94,8 +113,12 @@ func (c *IdleTimingConn) TimesOutIn() time.Duration {
 // Read implements the method from io.Reader
 func (c *IdleTimingConn) Read(b []byte) (int, error) {
 	c.closeMutex.RLock()
-	defer c.closeMutex.RUnlock()
```

Comment:
This was a drive-by. Defers are actually quite expensive, so doing them on the critical path isn't great. This addresses that.

Reply:
Nice catch!

Reply:
It was thanks to your benchmarking!